extension.js

```diff
@@ -356,10 +372,29 @@ function update_height_setting(win) {
     settings.set_double('window-height', Math.min(1.0, current_height));
 }
 
+function set_window_maximize() {
+    if (!current_window)
+        return;
+
+    const should_maximize = settings.get_boolean('window-maximize');
+    if (current_window.maximized_vertically && !should_maximize)
```

Comment:
You can simplify things by `if (current_window.maximized_vertically === should_maximize) return;`. And then just `if (should_maximize) maximize(); else unmaximize();`